api/routes/users/put.endpoint.js

```diff
@@ -0,0 +1,115 @@
+const {
+  getDB,
+  getFullPath,
+  login,
+  request,
+  unauthenticatedTest,
+  unauthorizedTest
+} = require('../../utils.endpoint');
+
+const url = userID => getFullPath(`/users/${userID}`);
+
+describe('users endpoint | PUT /users/:userID', async () => {
+  const db = getDB();
+  beforeAll(() => db.seed.run());
+  afterAll(() => db.destroy());
+
+  unauthenticatedTest('put', url(0));
+  unauthorizedTest('put', url(0));
+
+  describe('when authenticated', async () => {
+    let cookies;
+    beforeAll(async () => {
+      cookies = await login();
+    });
+    const put = (id, body) =>
+      request.put(url(id), {
+        jar: cookies,
+        json: body
+      });
+
+    it('when sending a non-numeric ID', async () => {
+      const { response, body } = await put('abc');
+
+      expect(response.statusCode).toEqual(400);
+      expect(body).toMatchSnapshot();
+    });
+
+    it('when updating a non-existant user ID', async () => {
+      const { response, body } = await put(0);
+
+      expect(response.statusCode).toEqual(404);
+      expect(body).toMatchSnapshot();
+    });
+
+    it('when attempting to update self', async () => {
+      const { response, body } = await put(2000);
+
+      expect(response.statusCode).toEqual(403);
+      expect(body).toMatchSnapshot();
+    });
+
+    describe('when updating a valid user ID', async () => {
+      it('...with an invalid state ID', async () => {
+        const { response, body } = await put(2001, { state: 'xx' });
+
+        expect(response.statusCode).toEqual(400);
```

Comment:
Why would we want a valid user ID to return a 400?

Reply:
The user is valid, but the state is not. AFAIK, 400 is a generic "the client sent something wrong" response, but if that's not right, happy to change it to something more appropriate